fastlane/lib/fastlane/actions/create_keychain.rb

```diff
@@ -38,34 +38,35 @@ def self.run(params)
             Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN] = Fastlane::Actions.sh("security default-keychain", log: false).strip
           rescue
           end
-          commands << Fastlane::Actions.sh("security default-keychain -s #{keychain_path}", log: false)
+          commands << Fastlane::Actions.sh("security default-keychain -s #{escaped_keychain_path}", log: false)
         end
 
-        commands << Fastlane::Actions.sh("security unlock-keychain -p #{escaped_password} #{keychain_path}", log: false) if params[:unlock]
+        commands << Fastlane::Actions.sh("security unlock-keychain -p #{escaped_password} #{escaped_keychain_path}", log: false) if params[:unlock]
 
         command = "security set-keychain-settings"
         command << " -t #{params[:timeout]}" if params[:timeout]
         command << " -l" if params[:lock_when_sleeps]
         command << " -u" if params[:lock_after_timeout]
-        command << " #{keychain_path}"
+        command << " #{escaped_keychain_path}"
 
         commands << Fastlane::Actions.sh(command, log: false)
 
         if params[:add_to_search_list]
           keychains = Action.sh("security list-keychains -d user").shellsplit
+          # Do not use 'escaped_keychain_path' here because .shelljoin will cause double escaping
```

Comment:
all the changes in this file are just so this is possible, correct?

Reply:
I did this to be more explicit with what is being used throughout the file.  I.e. `escaped_keychain_path` vs sprinkling `.shellescape` in all of these cases except for the `.shelljoin` case where we want to use the plain `keychain_path`.

Reply:
I did this to be more explicit with what is being used throughout the file.  I.e. `escaped_keychain_path` vs sprinkling `.shellescape` in all of these cases except for the `.shelljoin` case where we want to use the plain `keychain_path`.